### DISCUSSION TITLE
New provider initiate function for multiple chain RPC providers

### DISCUSSION THREAD

**Discussion by sk1122** [wevm/viem#288] [Idea / Feature Request] [answered]
Can we have a function like `createBatchedProvider` which will take list of `Provider`, each can either be same chain or multiple chain

We can then based on chain, route the RPC call and if that RPC call fails on 429 or 500, reroute it through other RPC of the same chain from the list

Or else, we can accept only same chain RPC providers and load balance call between them, like [Svinge](https://github.com/sk1122/svinge)

**Comment by jxom** [comment, marked as answer]
Check out: https://viem.sh/docs/clients/transports/fallback.html
